gbif/gbif_name_match/gbif_species_name_match.R

```diff
@@ -5,6 +5,12 @@ library('assertthat')
 library('assertable')
 
 
+API_terms = c('usageKey', 'scientificName', 'canonicalName', 'rank',
```

Comment:
This is not good practice. Set this in the function arguments. That makes it more clear for the user. And will show up in the documentation (when adding to a package)

Reply:
I would rather not define the whole list in the argument, as we have a preferred set of defaults (the list as it is now). However, for checking if arguments make sense, the check against the entire list is required.

Reply:
In he typical use case of `match.arg()` all valid options are listed in the function argument and one the first one is selected by default.
It is now clear to me that you have a different use case. So `match.args(gbif_terms, api_terms)` is OK. However the documentation should be updated to reflect that.

Reply:
ok, thanks. Interesting functionality indeed. I've updated the docs as well.

Reply:
Another cool thing is that `match.arg()` allows abbreviations.
```
test <- function(x = c("one", "two", "other")){
  x <- match.arg(x)
  cat(x)
}
test("t")
# two
test("o")
# Error in match.arg(x) : 'arg' should be one of “one”, “two”, “other”
test("ot")
# other
```